contracts/staking/ServiceStakingBase.sol

```diff
@@ -435,20 +440,81 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
         }
     }
 
+    /// @dev Evicts services due to their extended inactivity.
+    /// @param evictServiceIds Service Ids to be evicted.
+    function _evict(uint256[] memory evictServiceIds) internal {
+        uint256 totalNumServices = evictServiceIds.length;
+        uint256 numEvictServices;
+
+        // Get the number of evicted services
+        for (uint256 i = 0; i < totalNumServices; ++i) {
+            if (evictServiceIds[i] > 0) {
+                numEvictServices++;
+            }
+        }
```

Comment:
The input array is always of a size of all the staked services, and thus for the event exact number of evicted services must be calculated.